src/codeu/chat/client/maingui/MainGui.java

```diff
@@ -0,0 +1,77 @@
+package codeu.chat.client.maingui;
+
+import java.awt.*;
+import javax.swing.*;
+import javax.swing.border.Border;
+
+import codeu.chat.client.ClientContext;
+import codeu.chat.client.Controller;
+import codeu.chat.client.View;
+import codeu.chat.util.Logger;
+import codeu.chat.common.LoginInputCallback;
+
+// Chat - top-level client application - Java Simple GUI (using Java Swing)
+public final class MainGui implements LoginInputCallback{
+
+  private final static Logger.Log LOG = Logger.newLog(MainGui.class);
+
+  private JFrame mainFrame;
+
+  private final ClientContext clientContext;
+
+  // Constructor - sets up the Chat Application
+  public MainGui(Controller controller, View view) {
+    clientContext = new ClientContext(controller, view);
+  }
+
+  // Run the GUI client
+  public void run() {
+
+    try {
+
+      initialize();
+      mainFrame.setVisible(true);
+
+    } catch (Exception ex) {
+      System.out.println("ERROR: Exception in ChatSimpleGui.run. Check log for details.");
+      LOG.error(ex, "Exception in ChatSimpleGui.run");
+      System.exit(1);
+    }
+  }
+
+  private Border paneBorder() {
+    Border outside = BorderFactory.createLineBorder(Color.LIGHT_GRAY);
+    Border inside = BorderFactory.createEmptyBorder(8, 8, 8, 8);
+    return BorderFactory.createCompoundBorder(outside, inside);
+  }
+
+  // Initialize the GUI
+  private void initialize() {
+
+    // Outermost frame.
+    // NOTE: may have tweak size, or place in scrollable panel.
```

Comment:
grammar nits:
s/may have tweak/may have to tweak
s/in scrollable/in a scrollable